cmd/crank/render/runtime_docker.go

```diff
@@ -312,30 +312,39 @@ func (r *RuntimeDocker) createContainer(ctx context.Context, cli *client.Client)
 
 // startContainer ensures the container is running and returns its address.
 func (r *RuntimeDocker) startContainer(ctx context.Context, cli *client.Client, containerID string) (string, error) {
+
 	// Start the container (idempotent - safe to call on running containers)
 	if err := cli.ContainerStart(ctx, containerID, container.StartOptions{}); err != nil {
 		return "", errors.Wrap(err, "cannot start Docker container")
 	}
 
-	// Inspect the container to get the actual allocated port
+	// Inspect the container to get the actual allocated port.
 	inspect, err := cli.ContainerInspect(ctx, containerID)
 	if err != nil {
 		return "", errors.Wrap(err, "cannot inspect Docker container")
 	}
 
-	// Look up the specific function port instead of taking the first one
-	p := nat.Port(fmt.Sprintf("%d/tcp", FunctionPort))
-	if len(inspect.NetworkSettings.Ports[p]) == 0 {
-		return "", errors.Errorf("container %q has no published binding for port %s", r.Name, p.Port())
+	// Extract the allocated port from the container's port bindings.
+	// Iterate through all ports to find the first one with bindings.
+	var bindings []nat.PortBinding
+	for _, portBindings := range inspect.NetworkSettings.Ports {
+		if len(portBindings) > 0 {
+			bindings = portBindings
+			break
+		}
+	}
```

Comment:
Why is this change necessary? Seems like we could select the wrong port, since the first port with bindings could be something other than the `FunctionPort`.

Reply:
It seems more accurate to look up the specific `FunctionPort` that we care about. Did you find in your testing that that approach did not work? 

It would be nice to only change what's really needed here to fix your scenario, e.g. if we can make this PR only a change from `""` --> `"0"`, that would be ideal.

Reply:
at the same time, i wonder if this is the change that is actually making a difference and the `""` --> `"0"` change is a red herring. Could it be that the look up by `FunctionPort` doesn't work as reliably as we expect, and this approach of "just give me the first one that exists" works more reliably? That doesn't seem intuitive, but since @kaessert couldn't find the "async vs. sync" claim documented anywhere, I'm starting to wonder...